### ISSUE TITLE
python-trio false alarms

### ISSUE THREAD

**Issue by jakkdl** [Zac-HD/flake8-trio#37]
Quickly ran the plugin against python-trio: (although this includes tests and stuff, so a bunch may come from there)
```
$ for i in {100..112}; do echo -n $i": "; flake8 --select=TRIO$i | wc -l; done
100: 0
101: 0
102: 18
103: 11
104: 5
105: 0
106: 6
107: 276
108: 42
109: 1
110: 1
111: 0
112: 0
```
A huge number of 107, I suspect a lot of them are false positives that could mayhaps get filtered out. E.g. currently 107 doesn't check if an async function is used in a trio context at all - could maybe at least check if `trio` is imported in the file. Or a more sophisticated check would be if the function is referenced in a function that's in a trio context ... building up a huge tree up to `trio.run()`. Feels possible complicated or error-prone, but maybe worth it?

**Comment by Zac-HD** [comment]
I skimmed through all the non-test, non-107/108 warnings, and they all looked reasonable on both sides.

This does suggest that we might want to rename 107 and 108, to make it easier for the many projects I expect are just going to disable them.  Might also be useful to split three ways, 1xx codes for likely bugs, 2xx for idiom, 3xx for missing checkpoints.  Let's finish everything else first and then I'll decide what to do there.

**Comment by Zac-HD** [comment]
Looking over our warning codes, I don't think there's a nice clean division between bugs and idiom issues, so let's just leave them alone.
